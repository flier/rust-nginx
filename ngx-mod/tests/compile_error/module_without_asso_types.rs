use ngx_mod::{rt::ffi, Module};

#[derive(Module)]
struct M;

impl ngx_mod::Module for M {}

impl ngx_mod::http::Module for M {}

#[no_mangle]
static mut ngx_m_module_commands: [ffi::ngx_command_t; 0] = [];

fn main() {}
