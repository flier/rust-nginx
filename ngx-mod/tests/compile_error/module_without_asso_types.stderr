error[E0046]: not all trait items implemented, missing: `Error`, `MainConf`, `SrvConf`, `LocConf`
 --> tests/compile_error/module_without_asso_types.rs
  |
  | impl ngx_mod::http::Module for M {}
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ missing `Error`, `MainConf`, `SrvConf`, `LocConf` in implementation
  |
  = help: implement the missing item: `type Error = /* Type */;`
  = help: implement the missing item: `type MainConf = /* Type */;`
  = help: implement the missing item: `type SrvConf = /* Type */;`
  = help: implement the missing item: `type LocConf = /* Type */;`
